core/src/main/java/org/infinispan/metrics/impl/VendorAdditionalMetrics.java

```diff
@@ -32,12 +34,15 @@ public void bindTo(MeterRegistry registry) {
       List<MemoryPoolMXBean> memoryPoolBeans = ManagementFactory.getMemoryPoolMXBeans();
       for (MemoryPoolMXBean memoryPoolBean : memoryPoolBeans) {
          String name = memoryPoolBean.getName();
-         Gauge.builder(PREFIX + "memoryPool." + name + ".usage", memoryPoolBean, (mem) -> mem.getUsage().getUsed())
+         // avoid illegal characters due to beans named for example: "CodeHeap 'non-nmethods'"
+         String metricName = NameUtils.filterIllegalChars(name).toLowerCase();
```

Comment:
The lower case in some cases could be a problem.
For instance I think that `CodeCache` is converted into `codecache`,  losing the information about the fact that code and cache are two different sentences.

Reply:
The updated PR no longer call `toLowerCase()`